### ISSUE TITLE
Option to ignore words

### ISSUE THREAD

**Issue by WickyNilliams** [streetsidesoftware/vscode-spell-checker#146]
There are some words that I do not wish to be identified as misspellings. I could, of course, add these to the dictionary which would stop the warnings. But then these words would be offered as suggestions for other misspellings - which I do not want. Therefore an option to ignore some words is required.

**Comment by Jason3S** [comment]
Thank you for bringing this up. The setting exists, it is just not well documented. At the moment, there isn't any UI to add words to the ignore list.

There are three possible ways:
1. Add it in your document that is being checked:
    ```javascript
    // cspell:ignore colourings
    ```
2. In the Workspace settings: `cSpell.ignoreWords`.
    Example:
    ```json
    "cSpell.ignoreWords": ["colourings"]
    ```
3. In `cspell.json` located in the root of your workspace:
    It is:
    ```json
    "ignoreWords": ["colourings"]
    ```

**Comment by WickyNilliams** [comment]
Excuse the delay. Good to know it already exists :) Do you plan on adding some UI for this?

**Comment by Jason3S** [comment]
I will add it as a F1 command.
I will need to think about how to add it to the right-click menu. Since it is not configurable via an api, it is hard to give people the option to use it or not.

**Comment by WickyNilliams** [comment]
Why is it not configurable via API?

**Comment by Jason3S** [comment]
In VS Code (as of a couple of months ago), you have to define the menu items in advance via the package.json file.

```
    "contributes": {
        "menus": {
            "editor/context": [...]
        }
    }
```

Which makes it hard to add dynamic items to the context menu.

**Comment by WickyNilliams** [comment]
Hmm... perhaps I'm missing something, but how is ignoring any different to "Add: [word] to dictionary" actions? They seem dynamic?

**Comment by ChildishGiant** [comment]
Is there any update on adding this to right click?

**Comment by Spenhouet** [comment]
I don't like the "ignore by list in the settings" approach.
I would like the exact same workflow as the global dictionary provides.
The only difference that the ignore words are a seperate list of the dictionary and are not used for suggestions.

**Comment by LaurentOngaro** [comment]
Please can you add this options. 
It's very painfull to edit settings.json just to add ignored words. 
Many thanks

**Comment by courtneyaschwartz** [comment]
@Jason3S  Sorry, is this available? (Issue is "Open", but the label is "Done/Fixed"...?)

**Comment by courtneyaschwartz** [comment]
Not sure if this will be the fix I'm looking for. But basically instead of this:
`"ignoreWords" : ["hte" //plus a hundred others]`

I need like this:
```
"ignoreWordsDictionary": "ignored-words-dict",
"ignoreWordsDictionaryDefinitions": [ {"name": "ignored-words-dict", "path": "./cspell_ignored.txt"}]

```
(Re-use an external ignore list across multiple projects, I guess like Spenhouet. Don't want to suggest my ignored list as corrections, hence not loaded as a regular `dictionary`.)

Is something like this being added? Or just a UI for `ignoreWords`...?

**Comment by Jason3S** [comment]
@courtneyaschwartz At the moment there is only UI to add it to the `ignoreWords` list.

I agree, a dictionary option would be very useful.

**Comment by ky-huang** [comment]
I'm glad that there's an 'Ignore Word' command in right click menu now. But I feel it inconvenient that I have to press enter after 'Ignore Word'... Maybe removing this 'press enter' step is better, because right click won't causs mistakes often so it might be unnecessary to require a confirmation. Even if you ignore a word by mistake, you can delete the word in `.vscode/settings.json`.

**Comment by moreati** [comment]
> But I feel it inconvenient that I have to press enter after 'Ignore Word'

I'm finding "press enter to confirm" is skipped if I select the word before right clicking.

**Comment by Jason3S** [comment]
@moreati,

There is a new version coming out soon. The behavior of the right-click menu has changed. 
Please try it out: [Releases](https://github.com/streetsidesoftware/vscode-spell-checker/releases).

**Comment by Jason3S** [comment]
I'm going to close this as "fixed". Documentation could be better, but it is possible to ignore words. It is even possible to setup a dictionary of words to be ignored, but not suggested.
